tracer/test/Datadog.Trace.TestHelpers/TracerHelper.cs

```diff
@@ -15,14 +17,36 @@ internal static class TracerHelper
         /// <summary>
         /// Create a test instance of the Tracer, that doesn't used any shared instances
         /// </summary>
-        public static Tracer Create(
+        public static ScopedTracer Create(
             TracerSettings settings = null,
             IAgentWriter agentWriter = null,
             ITraceSampler sampler = null,
             IScopeManager scopeManager = null,
             IDogStatsd statsd = null)
         {
-            return new Tracer(settings, agentWriter, sampler, scopeManager, statsd);
+            return new ScopedTracer(settings, agentWriter, sampler, scopeManager, statsd);
+        }
+
+        /// <summary>
+        /// Create a test instance of the Tracer, that doesn't used any shared instances
+        /// </summary>
+        public static ScopedTracer CreateWithFakeAgent(
+            TracerSettings settings = null)
+        {
+            return new ScopedTracer(settings, Mock.Of<IAgentWriter>());
+        }
+
+        public class ScopedTracer : Tracer, IDisposable
+        {
+            public ScopedTracer(TracerSettings settings = null, IAgentWriter agentWriter = null, ITraceSampler sampler = null, IScopeManager scopeManager = null, IDogStatsd statsd = null)
+                : base(settings, agentWriter, sampler, scopeManager, statsd)
+            {
+            }
+
+            public void Dispose()
+            {
+                TracerManager.ShutdownAsync().GetAwaiter().GetResult();
+            }
```

Comment:
Shame we can't use `DisposeAsync()` because of .NET FX 🙁

Reply:
Yeah 😞

Reply:
Though maybe we can shim it 🤔